Src/FluentAssertions/Equivalency/DataRowCollectionEquivalencyStep.cs

```diff
@@ -0,0 +1,241 @@
+﻿using System;
+using System.Data;
+using System.Linq;
+
+using FluentAssertions.Data;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Equivalency
+{
+    public class DataRowCollectionEquivalencyStep : IEquivalencyStep
+    {
+        public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAssertionOptions config)
+        {
+            return typeof(DataRowCollection).IsAssignableFrom(config.GetExpectationType(context));
+        }
+
+        public bool Handle(IEquivalencyValidationContext context, IEquivalencyValidator parent, IEquivalencyAssertionOptions config)
```

Comment:
❌ This method must be refactored. It's just too long to review.

Reply:
I have done this and it will be in the next commit.